### PR TITLE
Docker Finalization

### PR DESCRIPTION
**Pull request by rahul-tuli**
This  PR will update our GHA and dockerfiles:
  - [X] Docker Images should be built and pushed whenever a release is published (Update GHA workflow trigger)
  - [X] Dockerfile updates to install from pypi and support for nightlies
  - [X] Image naming updates, GHA workflow file

The Dockerfile also includes support to build a dev container with the following command within `docker` directory:
```bash
DOCKER_BUILDKIT=1 \
  docker build  \
  --target dev \
  --build-arg DEPS=dev \
  --build-arg BUILDKIT_INLINE_CACHE=1 \
  -t sparseml:dev .
```

The Images will be pushed to ghcr.io and will be available under pushed packages for the repo

### CODE REVIEW

**Review by KSGulin — DISMISSED** [review]
Looks good! I like the use of console prints for step/progress tracking

File: docker/Dockerfile [hunk 1]
```diff
@@ -12,13 +12,6 @@ FROM nvidia/cuda:${CUDA_VERSION}-devel-ubuntu18.04 as base
 # Source: https://github.com/NVIDIA/nvidia-docker/issues/1632
 
 ARG VENV
-
-# metainformation
-LABEL org.opencontainers.image.version = "1.4"
-LABEL org.opencontainers.image.authors = "Neuralmagic, Inc."
-LABEL org.opencontainers.image.source = "https://github.com/neuralmagic/sparseml"
-LABEL org.opencontainers.image.licenses = "Apache License 2.0"
```
Comment (corey-nm) [inline docker/Dockerfile:20]: Do we not need this anymore?

File: .github/workflows/build-and-publish-release-images.yaml [hunk 2]
```diff
@@ -52,68 +53,46 @@ jobs:
...
-        uses: docker/build-push-action@v2
-        with:
-          context: ./docker
           push: true
-          build-args: |
-            DEPS=yolov5
-            VERSION=${{ steps.extract_tag.outputs.tag }}
           tags: |
-            ghcr.io/neuralmagic/sparseml-yolov5:${{ steps.extract_tag.outputs.tag }}
+            ghcr.io/neuralmagic/sparseml-base:${{ steps.extract_tag.outputs.tag }}
```
Comment (corey-nm) [inline .github/workflows/build-and-publish-release-images.yaml:81]: Thoughts on adding `+cu11.1.1` here similar to the 10.2 ones?

**Review by corey-nm — APPROVED** [review]
lgtm, nice work! 👍

File: .github/workflows/build-and-publish-release-images.yaml (hunk 2 above)
Comment (rahul-tuli) [reply .github/workflows/build-and-publish-release-images.yaml:81]: Cuda 11.1.1 will be the default version we target and decided only to push extra info on tag name for cuda 10.2

**Review by corey-nm — APPROVED** [review]

**Review by KSGulin — APPROVED** [review]

File: docker/Dockerfile (hunk 1 above)
Comment (rahul-tuli) [reply docker/Dockerfile:20]: No this was added by an external user, it's good but adds one more thing we'll have to maintain, the tags pushed to ghcr already take care of the image version

**Review by bfineran — APPROVED** [review]

**Pull request merged**